### PR TITLE
Notices Redux: isPersistent and displayOnNextPage options

### PR DESCRIPTION
**Pull request by artpi**
This PR continues work from #1496.

## What for?

It makes notices non permanent by default

It adds 2 properties to notices options that were present in previous notices implementation
- `isPersistent`, which makes the notice appear after reload
- `displayOnNextPage`, which displays notice on next page display

## Testing

- Go to `/me/notifications`
- Change notification settings for a site
- See that success notice disappears on route change

### CODE REVIEW

File: client/me/notification-settings/index.jsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ const NotificationSettings = React.createClass( {
 		}
 
 		if ( state.status === 'success' ) {
-			this.props.successNotice( this.translate( 'Settings saved successfully!' ), { duration: 3000 } );
+			this.props.successNotice( this.translate( 'Settings saved successfully!' ) );
```
Comment (rralian) [inline client/me/notification-settings/index.jsx]: Why are you removing the duration here?

Comment (rralian) [inline client/me/notification-settings/index.jsx]: I guess we're relying upon route change to clear the notice instead of time now, but that's a functional change that I think should at least be mentioned/discussed. FWIW I see no problem with it though.

**Comment by rralian** [comment]
LGTM! :shipit:

File: client/me/notification-settings/index.jsx (hunk 1 above)
Comment (artpi) [inline client/me/notification-settings/index.jsx]: I added this duration here in #1496, previously it was not there :)

File: client/state/notices/actions.js [hunk 2]
```diff
@@ -38,6 +41,13 @@ function createNotice( status, text, options = {} ) {
 	}
 }
 
+export function setRoute( path ) {
```
Comment (aduth) [inline client/state/notices/actions.js]: This doesn't appear to be used anywhere? Also, seems unrelated to notices, so while it might make sense for the notices reducer to hook in to the action type, I think the action should probably live elsewhere (in a potential routing subtree, perhaps?).

Comment (artpi) [inline client/state/notices/actions.js]: It is, for the notice to disappear when navigating to new route
https://github.com/Automattic/wp-calypso/pull/2009/files#diff-77b3f6be0290b9520a2b10b906721494R23

It is triggered in `boot/index`:
https://github.com/Automattic/wp-calypso/pull/2009/files#diff-77b3f6be0290b9520a2b10b906721494R23

Comment (artpi) [inline client/state/notices/actions.js]: And I agree, I would like this to be somewhere else, but first we need to tackle routing in redux, which I would like to see :)
But for now, it makes sense here IMO. If we have more use cases for this kind of action, I will switch to the proper action from a proper tree.

Comment (aduth) [inline client/state/notices/actions.js]: >It is triggered in `boot/index`:

Ah, search failed me, I was looking for `setRoute(` . Thanks!

>but first we need to tackle routing in redux, which I would like to see :) But for now, it makes sense here IMO.

I don't think we need to solve the routing problem for this to be moved. Why not just create `client/state/routing/actions.js` and drop this function there?
